mlflow/data/spark_dataset.py

```diff
@@ -130,6 +130,11 @@ def profile(self) -> Optional[Any]:
                 high=approx_count_result.high(),
             )
             approx_count = int(approx_count_float)
+            if approx_count <= 0:
+                # An approximate count of zero likely indicates that the count timed
+                # out before an estimate could be made. In this case, we use the value
+                # "unknown" so that users don't think the dataset is empty
+                approx_count = "unknown"
```

Comment:
I considered returning an empty dict for `approx_count <= 0`, but I thought that it might be confusing / look like a bug if some spark DFs have the approx count and others don't. I think it's best to tell the user that we tried to get the count and weren't able to determine it